spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/LettuceConnectionConfiguration.java

```diff
@@ -91,27 +95,38 @@ private LettuceConnectionFactory createLettuceConnectionFactory(LettuceClientCon
 
 	private LettuceClientConfiguration getLettuceClientConfiguration(
 			ObjectProvider<LettuceClientConfigurationBuilderCustomizer> builderCustomizers,
-			ClientResources clientResources, Pool pool) {
-		LettuceClientConfigurationBuilder builder = createBuilder(pool);
+			ClientResources clientResources, Pool pool, Environment environment) {
+		LettuceClientConfigurationBuilder builder = createBuilder(environment, pool);
+
 		applyProperties(builder);
 		if (StringUtils.hasText(getProperties().getUrl())) {
 			customizeConfigurationFromUrl(builder);
 		}
 		builder.clientOptions(createClientOptions());
 		builder.clientResources(clientResources);
+
 		builderCustomizers.orderedStream().forEach((customizer) -> customizer.customize(builder));
 		return builder.build();
 	}
 
-	private LettuceClientConfigurationBuilder createBuilder(Pool pool) {
-		if (pool == null) {
-			return LettuceClientConfiguration.builder();
+	private LettuceClientConfigurationBuilder createBuilder(Environment environment, Pool pool) {
+		final boolean poolEnabled = environment.getProperty("spring.redis.lettuce.pool.enabled", Boolean.class, true);
+		if (poolEnabled) {
+			evaluatePoolingFailureAnalyzer();
+			return new PoolBuilderFactory().createBuilder(pool);
+		}
+		return LettuceClientConfiguration.builder();
+	}
+
+	private void evaluatePoolingFailureAnalyzer() {
+		final boolean pool2Present = ClassUtils.isPresent(PoolBuilderFactory.GENERIC_OBJECT_POOL_CONFIG_CLASS_NAME,
+				null);
+		if (!pool2Present) {
+			throw new RedisClientPoolingException(ClientType.LETTUCE);
```

Comment:
There's no need for that. If the pool is not available, then no pool should be configured as before.